pytorch_lightning/trainer/training_loop.py

```diff
@@ -661,20 +661,12 @@ def run_training_batch(self, batch, batch_idx, dataloader_idx):
         splits = self.tbptt_split_batch(batch)
 
         for split_idx, split_batch in enumerate(splits):
-            self.trainer.split_idx = split_idx
 
-            # in manual optimization we loop over all optimizers at once
-            optimizers = self.get_optimizers_iterable()
-            if not self.automatic_optimization:
-                optimizers = [optimizers[0]]
+            # create an iterable for optimizers and loop over them
+            for opt_idx, optimizer in self.prepare_optimizers():
```

Comment:
Is each item here already a tuple or do we have to enumerate here?

Reply:
There are already a tuple.